lib/hammer_cli_foreman/api/connection.rb

```diff
@@ -1,22 +1,33 @@
 require 'hammer_cli_foreman/api/session_authenticator_wrapper'
 require 'hammer_cli_foreman/api/interactive_basic_auth'
+require 'hammer_cli_foreman/api/oauth/interactive_keycloak_auth_via_2fa'
+require 'hammer_cli_foreman/api/oauth/interactive_keycloak_auth'
 require 'hammer_cli_foreman/api/void_auth'
 
 module HammerCLIForeman
+  AUTH_TYPES = {
+    basic_auth: "Basic_Auth",
+    oauth_with_2fa: "Oauth_With_Two_Factor",
+    oauth_without_2fa: "Oauth_Without_Two_Factor"
+  }.freeze
+
+  CONNECTION_NAME = 'foreman'
+
   module Api
     class Connection < HammerCLI::Apipie::ApiConnection
       attr_reader :authenticator
 
-      def initialize(settings, logger = nil, locale = nil)
-        default_params = build_default_params(settings, logger, locale)
+      def initialize(settings, logger = nil, locale = nil, auth_type =
+          AUTH_TYPES[HammerCLI::Settings.get(:foreman, :default_auth_type).to_sym])
```

Comment:
if the settings does not contain the :default_auth_type I get 
`NoMethodError (undefined method `to_sym' for nil:NilClass)`. We will need some default value here.

Reply:
Yeah, it is a known issue. I figured that out :) Need to fix this!